### PR TITLE
Load dataset from disc - dgx

### PR DESCRIPTION
**Pull request by ciebson123**
# Description
Loading datasets from common disc
For now, works only on dgx with c4 dataset

Neptune to show that it loads: https://app.neptune.ai/o/pmtest/org/llm-random/runs/compare?viewId=9b436221-e38b-45cb-9844-e89315a6fe42&detailsTab=metadata&shortId=LLMRANDOM-7166&dash=charts&type=run&chartFilter=loss&compare=OwRgbMA0BMkkA

# Checklist
   - [x] Is this PR focused on a single feature without the possibility of being divided into smaller PRs?
   - [x] Are all variables set with logical defaults that are backward compatible?
   - [x] I attached link to neptune if it was necessary
   - [x] Have you successfully executed the linter and tests?
   - [x] Are all functions concise and don't require splitting?
   - [x] Is there documentation for arguments and complex logic?
   - [x] Are there no temporary solutions in the code? Should a task be added to Trello to refine the code?
   - [x] Are all functions placed in appropriate files?
   - [x] Is the PR name meaningful and descriptive?
   - [x] Is PR description provided, or unnecessary?
  
Other questions:
   - [x] If I made changes to the requirements or anything related to the Singularity image I did generate new image and upload a new image to the clusters
   - [x] If this change is significant enough I notified all individuals working with this repository (@channel in slack channel)
   - [x] If there is a need for a second reviewer I requested additional review 

# Reviewer's checklist:
   - [ ] Does the Neptune link work and does it actually compare the situation before and after the PR?
   - [ ] Do I understand the code?
   - [ ] Do I think another reviewer is not needed or I commented that this PR need another reviewer?

### CODE REVIEW

File: lizrd/scripts/grid.py [hunk 1]
```diff
@@ -123,6 +126,10 @@
             neptune_key=args.neptune_key,
         )
 
+        singulatility_mount_paths = make_singularity_mount_paths(
```
Comment (maciek-pioro) [inline lizrd/scripts/grid.py:129]: typo

File: research/conditional/utils/argparse.py [hunk 2]
```diff
@@ -60,6 +60,9 @@ def introduce_parser_arguments(
     # other data hyperparameters
     parser.add_argument("--num_workers", type=int, default=8)
 
+    parser.add_argument("--train_dataset_path", type=str, default=None)
+    parser.add_argument("--validation_dataset_path", type=str, default=None)
```
Comment (maciek-pioro) [inline research/conditional/utils/argparse.py:64]: Can we add some help for that? e.g. IIUC this only works for the C4 dataset?

**Review by maciek-pioro — CHANGES_REQUESTED** [review]
Some small comments, but apart from that - good job, a very useful change.

File: lizrd/scripts/grid.py (hunk 1 above)
Comment (ciebson123) [reply lizrd/scripts/grid.py:129]: I new it was somewhat too big

**Review by maciek-pioro — APPROVED** [review]
